src/docfx/template/TemplateEngine.cs

```diff
@@ -54,14 +54,14 @@ internal class TemplateEngine : IDisposable
             _mustacheTemplate = new MustacheTemplate(_contentTemplateDir, _global, jsonSchemaTransformer);
         }
 
-        public bool IsHtml(ContentType contentType, string? mime)
+        public RenderType GetRenderType(ContentType contentType, SourceInfo<string?> mime)
         {
             return contentType switch
             {
-                ContentType.Redirection => true,
-                ContentType.Page => IsConceptual(mime) || IsLandingData(mime) || _mustacheTemplate.HasTemplate($"{mime}.html"),
-                ContentType.TableOfContents => _mustacheTemplate.HasTemplate($"toc.html"),
-                _ => false,
+                ContentType.Redirection => RenderType.Content,
+                ContentType.Page => (IsConceptual(mime) || IsLandingData(mime) || IsContentScheme(mime)) ? RenderType.Content : RenderType.Component,
```

Comment:
Eliminate the conversion to bool `RenderType` --> `bool` --> `RenderType`?. If it is not conceptual or landing data, return `GetSchema().RenderType`?

```suggestion
                ContentType.Page => GetPageRenderType(mime)
```